draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright [2022] [Application library engineering group]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+import Curves from '@ohos.curves'
+import { Cell } from './types'
+import { DraggableModel } from './modelClass'
+
+const springSetting1: string = Curves.spring(0, 1, 180, 10)
+const springSetting2: string = Curves.spring(0, 1, 120, 15)
+
+@Component
+export struct Draggable {
+  model: DraggableModel.Model = undefined
+  private mouseXY: number[] = [0, 0]
+  private positionOffsetCorrection: number[] = [0, 0]
+  @State private lastPress: number = undefined
+  @State private isPressed: boolean = false
+  @State private scale: number = 1
+  @State private order: number[] = undefined
+  private innerFlexSize: SizeOptions = undefined
+  private layout: Cell[] = undefined
+  private ballCount: number
+  private cellWidth: number
+  private cellHeight: number
+
+  aboutToAppear() {
+    this.ballCount = this.model.getColorsList().length
+    this.cellWidth = DraggableModel.Model.len2num(this.model.getBallSize().width) + 20
+    this.cellHeight = DraggableModel.Model.len2num(this.model.getBallSize().height) + 40
```

Comment:
Check the getBallSize(), getColorList if its undefined or null. Do this check wherever applicable

Reply:
values for these properties are passed as required parameters in the constructor. So even if the user misses to give value for any of these variables, the framework will automatically throw an error.

Reply:
DONE

Reply:
done